src/App.tsx

```diff
@@ -15,36 +16,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  SORT_FIELD_ALPHABET = 'alphabet',
+  SORT_FIELD_LENGTH = 'length',
+  SORT_FIELD_REVERSE = 'reverse',
+  SORT_FIELD_DEFAULT = '',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: string,
+  isReverse: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  preparedGoods.sort((good1, good2) => {
+    switch (sortField) {
+      case SortType.SORT_FIELD_ALPHABET:
+        return good1.localeCompare(good2);
+
+      case SortType.SORT_FIELD_LENGTH:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (sortField === SortType.SORT_FIELD_REVERSE) {
+    return preparedGoods.reverse();
+  }
+
+  if (isReverse) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState(SortType.SORT_FIELD_DEFAULT);
+  const [isReverse, setIsReverse] = useState(false);
+
+  const sortGoods = getPreparedGoods(goodsFromServer, sortField, isReverse);
+
+  const toggleReverse = () => {
+    setIsReverse(prevIsReverse => !prevIsReverse);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => setSortField(SortType.SORT_FIELD_ALPHABET)}
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.SORT_FIELD_ALPHABET,
+          })}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => setSortField(SortType.SORT_FIELD_LENGTH)}
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortType.SORT_FIELD_LENGTH,
+          })}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={toggleReverse}
+          type="button"
+          className={cn('button is-warning', { 'is-light': !isReverse })}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {sortField || isReverse ? (
+          <button
+            onClick={() => {
+              setSortField(SortType.SORT_FIELD_DEFAULT);
+              setIsReverse(false);
+            }}
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here